services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocolService.java

```diff
@@ -245,6 +246,13 @@ public Object issueUserInfo() {
         return endpoint;
     }
 
+    @Path("grants")
```

Comment:
As for the way of implementing Endpoint class, It might be preferable to follow `DeviceEndpoint` and `DeviceEndpointFactory` rather than implementing it in this service class.